src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -124,32 +124,55 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
+                using var cancellationTokenSource = new CancellationTokenSource();
+
+                context.CancellationToken = cancellationTokenSource.Token;
                 var testExecution = RunTestOnSeparateThread(context);
-                if (Task.WaitAny(new Task[] { testExecution }, _timeout) != -1
```

Comment:
question: I know this builds off the existing code pattern to use `Task.WaitAny(Task[], int)`, but since we're adding a CTS here would it make sense to instead use `CTS.CancelAfter(int)` here?